src/Core/Services/AirtelMoneyService.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Text.RegularExpressions;
+using Core.Entities;
+
+namespace Core.Services
+{
+    public class AirtelMoneyService : IPaymentService
+    {
+        public readonly string _message;
+        public readonly string _phoneNumber;
+
+        public AirtelMoneyService(string message, string phoneNumber)
+        {
+            _message = IPaymentService.SanitizeMessage(message);
+            _phoneNumber = phoneNumber;
+        }
+        public Payment GeneratePayment()
+        {
+            var amountRegex = new Regex(@"((?<=((receivedMK)|(recievedMK)))(.*?)(?=from))");
+            var ger = amountRegex.Match(_message).ToString();
+            var amount = Decimal.Parse(amountRegex.Match(_message).ToString());
+            var reference = CreateRefrence();
+            var SenderName = Regex.Match(GetSenderName(), @"(?<=.*(?=[a-z]|[A-Z])).*").ToString();
+
+            return new Payment()
+            {
+                Amount = amount,
+                PhoneNumber = _phoneNumber,
+                AgentName = "Missing",
```

Comment:
are we sure there is absolutely no scenario where we can extract the agentName from an sms? 🤔

Reply:
As far as I know, there is no way of doing such. Its a problem on the airtel side